model/switch_state_test.go

```diff
@@ -0,0 +1,310 @@
+// Copyright 2022 The Serverless Workflow Specification Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	val "github.com/serverlessworkflow/sdk-go/v2/validator"
+)
+
+func TestSwitchStateStructLevelValidation(t *testing.T) {
+	type testCase struct {
+		desp string
+		obj  SwitchState
+		err  string
+	}
+	testCases := []testCase{
+		{
+			desp: "normal & eventConditions",
+			obj: SwitchState{
+				BaseState: BaseState{
+					Name: "1",
+					Type: "switch",
+				},
+				DefaultCondition: DefaultCondition{
+					Transition: &Transition{
+						NextState: "1",
+					},
+				},
+				EventConditions: []EventCondition{
+					{
+						EventRef: "1",
+						Transition: &Transition{
+							NextState: "2",
+						},
+					},
+				},
+			},
+			err: ``,
+		},
+		{
+			desp: "normal & dataConditions",
+			obj: SwitchState{
+				BaseState: BaseState{
+					Name: "1",
+					Type: "switch",
+				},
+				DefaultCondition: DefaultCondition{
+					Transition: &Transition{
+						NextState: "1",
+					},
+				},
+				DataConditions: []DataCondition{
+					{
+						Condition: "1",
+						Transition: &Transition{
+							NextState: "2",
+						},
+					},
+				},
+			},
+			err: ``,
+		},
+		{
+			desp: "missing eventConditions & dataConditions",
+			obj: SwitchState{
+				BaseState: BaseState{
+					Name: "1",
+					Type: "switch",
+				},
+				DefaultCondition: DefaultCondition{
+					Transition: &Transition{
+						NextState: "1",
+					},
+				},
+			},
+			err: `Key: 'SwitchState.SwitchState' Error:Field validation for 'SwitchState' failed on the 'required' tag`,
+		},
+		{
+			desp: "exclusive eventConditions & dataConditions",
+			obj: SwitchState{
+				BaseState: BaseState{
+					Name: "1",
+					Type: "switch",
+				},
+				DefaultCondition: DefaultCondition{
+					Transition: &Transition{
+						NextState: "1",
+					},
+				},
+				EventConditions: []EventCondition{
+					{
+						EventRef: "1",
+						Transition: &Transition{
+							NextState: "2",
+						},
+					},
+				},
+				DataConditions: []DataCondition{
+					{
+						Condition: "1",
+						Transition: &Transition{
+							NextState: "2",
+						},
+					},
+				},
+			},
+			err: `Key: 'SwitchState.SwitchState' Error:Field validation for 'SwitchState' failed on the 'exclusive' tag`,
```

Comment:
can't we change the fields for the validators? having the same field twice sounds odd and hard to identify what really is the required field, according to the validator it is the transition and end, right? so we should try to use it instead fieldA.fieldA.

Reply:
any fields name suggested？because this validation is for >1 fields, so i didn't find a clear & meaningful name for it.

Reply:
maybe "StructName.FieldName"